ledger/src/bigtable_upload.rs

```diff
@@ -126,42 +129,49 @@ pub async fn upload_confirmed_blocks(
         blocks_to_upload.last().unwrap()
     );
 
-    // Load the blocks out of blockstore in a separate thread to allow for concurrent block uploading
-    let (_loader_thread, receiver) = {
+    // Distribute the blockstore reading across a few background threads to speed up the bigtable uploading
+    let (_loader_threads, receiver): (Vec<_>, _) = {
         let exit = exit.clone();
 
         let (sender, receiver) = bounded(BLOCK_READ_AHEAD_DEPTH);
-        (
-            std::thread::spawn(move || {
-                let mut measure = Measure::start("block loader thread");
-                for (i, slot) in blocks_to_upload.iter().enumerate() {
-                    if exit.load(Ordering::Relaxed) {
-                        break;
-                    }
 
-                    let _ = match blockstore.get_rooted_block(*slot, true) {
-                        Ok(confirmed_block) => sender.send((*slot, Some(confirmed_block))),
-                        Err(err) => {
-                            warn!(
-                                "Failed to get load confirmed block from slot {}: {:?}",
-                                slot, err
-                            );
-                            sender.send((*slot, None))
+        let (slot_tx, slot_rx) = unbounded();
```

Comment:
```suggestion
        let (slot_sender, slot_receiver) = unbounded();
```
`tx` is overloaded in solana-land ;)